udata/migrations/2017-10-05-purge-void-resources.js

```diff
@@ -0,0 +1,23 @@
+/*
+ * Remove resources with missing required fields (url or title)
+ */
+
+var nbRemoved = 0;
+
+const removedNoTitle = db.dataset.update(
+    {resources: {$gt: []}},
```

Comment:
You can reduce the number of documents to lookup with `{resources.title: null}`.
If you want to perform it in a single pass:
```
{$or: [{'resources.title': null}, {'resources.url': null}]},
{$pull: {resources: {$or: [{title: null}, {url: null}]}}},
```